cli/cliui/cliui.go

```
@@ -117,48 +101,150 @@ func ifTerm(fmt pretty.Formatter) pretty.Formatter {
+	FocusedPrompt *userThemeStyle `toml:"focusedprompt"`
+	Keyword       *userThemeStyle `toml:"keyword"`
+	Placeholder   *userThemeStyle `toml:"placeholder"`
+	Prompt        *userThemeStyle `toml:"prompt"`
+	Warn          *userThemeStyle `toml:"warn"`
+}
+
+type userThemeConfig struct {
+	Colors map[string]string `toml:"colors"`
+	Styles *userThemeStyles  `toml:"styles"`
+}
+
+func applyUserTheme(theme userThemeConfig) {
+	if theme.Styles.Code != nil {
+		DefaultStyles.Code = pretty.Style{ifTerm(pretty.XPad(1, 1))}
+		DefaultStyles.Code = append(DefaultStyles.Code, theme.Styles.Code.toPrettyStyle(theme)...)
+	}
+	if theme.Styles.DateTimeStamp != nil {
+		DefaultStyles.DateTimeStamp = theme.Styles.DateTimeStamp.toPrettyStyle(theme)
+	}
+	if theme.Styles.Error != nil {
+		DefaultStyles.Error = theme.Styles.Error.toPrettyStyle(theme)
+	}
+	if theme.Styles.Field != nil {
+		DefaultStyles.Field = pretty.Style{pretty.XPad(1, 1)}
+		DefaultStyles.Field = append(DefaultStyles.Field, theme.Styles.Field.toPrettyStyle(theme)...)
+	}
+	if theme.Styles.Keyword != nil {
+		DefaultStyles.Keyword = theme.Styles.Keyword.toPrettyStyle(theme)
+	}
+	if theme.Styles.Warn != nil {
+		DefaultStyles.Warn = theme.Styles.Warn.toPrettyStyle(theme)
+	}
+	if theme.Styles.Placeholder != nil {
+		DefaultStyles.Placeholder = theme.Styles.Placeholder.toPrettyStyle(theme)
+	}
+	// NOTE: Prompt should be styled first as FocusedPrompt depends on the styling
+	// of Prompt.
+	if theme.Styles.Prompt != nil {
+		DefaultStyles.Prompt = theme.Styles.Prompt.toPrettyStyle(theme)
+		DefaultStyles.Prompt = append(DefaultStyles.Prompt, pretty.Wrap("> ", ""))
+	}
+	if theme.Styles.FocusedPrompt != nil {
+		style := theme.Styles.FocusedPrompt.toPrettyStyle(theme)
+		DefaultStyles.FocusedPrompt = append(DefaultStyles.Prompt, style...)
+	}
+}
+
+func (t userThemeConfig) getColor(color string) string {
+	if strings.HasPrefix(color, "$") {
+		if c, ok := t.Colors[color[1:]]; ok {
+			return c
+		}
```

Erroring here would be more user-friendly IMO, or what's the behavior now when you're referencing a non-existent color?

---

Currently if you provide an invalid colour it will silently ignore it, in `toPrettyStyle` it attempts to construct a `Color` and if this fails it does nothing. For similar reasons as before, maybe we should log an error here (providing we keep this colors section).

---

This is no longer an issue as the colors section has been removed